students/Dkuchan/session07/classmailroom.py

```diff
@@ -0,0 +1,263 @@
+#classmailroom.py
+
+# classessheet.py
+""" This is a script to mess with classes.
+    The idea is to use this practice to create class "persons" for mail room """
+# import os
+
+
+class Donor:
+    # This is an experimental class for donors in mailroom
+    def __init__(self, name):
+
+        self.first_name, self.last_name = list(name.split())
+        self.donations = [1, 2, 3, 4, 5, 6, 7]
+        # self.fullname = str(self.first_name + ' ' + self.last_name)
+        # self.averagedon = sum(self.donations) / len(self.donations)
+        # self.maxdon = max(self.donations)
+        # self.totaldonations = sum(self.donations)
+
+    @property
+    def totaldonations(self):
+        return(sum(self.donations))
+
+    @property
+    def maxdon(self):
+        return(max(self.donations))
+
+    @property
+    def averagedon(self):
+        return(self.totaldonations / len(self.donations))
+
+    @property
+    def namelen(self):
+        return(len(str(self.fullname)))
+
```

Comment:
This shouldn't be necessary. self.fullname should always be a string, and so len(self.fullname) should be sufficient. Also, this really doesn't need to be a property, since it is only used in one place, and the user won't care about it. Just find it directly in the longestname function.